src/iop/tonecurve.c

```diff
@@ -344,6 +347,19 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
                                          : d->table[ch_b][CLAMP((int)(b_in * 0x10000ul), 0, 0xffff)]);
         }
       }
+      else if(autoscale_ab == DT_S_SCALE_MANUAL_LCH)
+      {
+        const float pi = 3.14159f;
+        const float C_in = sqrtf(in[1]*in[1] + in[2]*in[2]) / 181.019f;
+        const float h_in = ((in[1]!=0.0f ? atanf(in[2]/in[1]) : (in[2]>0.0f ? 0.5f : -0.5f) * pi) + ((in[1] < 0.0f) ? pi : ((in[2] < 0.0f) ? 2.0f * pi : 0.0f))) / (2.0f * pi);
+        float C_out = L_in == 0.0f ? C_in : C_in * d->table[ch_CL][CLAMP((int)(L_in * 0x10000ul), 0, 0xffff)] * 0.02f;
```

Comment:
Please write a color conversion function in the lib  `src/common/colorspaces_inline_conversion.h`

Reply:
I have a SSE one in progress:
```
static inline __m128 dt_Lab_to_Lch_sse2(const __m128 Lab)
{
  __m128 Lch = Lab;
  __m128 Lab2 = Lab * Lab;
  Lch[1] = powf((Lab2[1] + Lab2[2]), 0.5f);
  Lch[2] = atan2f(Lab[2], Lab[1]);
  return Lch;
}

static inline __m128 dt_Lch_to_Lab_sse2(const __m128 Lch)
{
  __m128 Lab = Lch;
  Lab[1] = cosf(Lch[2]) * Lch[1];
  Lab[2] = sinf(Lch[2]) * Lch[1];
  return Lab;
}
```

Reply:
Need more time to understand this.

Reply:
I've added open cl support as done in the module. It uses basic.cl and colorspaces.cl. 
It doesn't use SSE. Could be interesting to improve that.